torchgeo/datasets/geo.py

```diff
@@ -692,7 +715,7 @@ def __init__(
             match = re.match(filename_regex, os.path.basename(filepath))
             if match is not None:
                 try:
-                    with fiona.open(filepath) as src:
+                    with fiona.open(filepath, layer=layer) as src:
```

Comment:
@isaaccorley you were talking about replacing fiona with geopandas in a future PR. Does geopandas have a similar feature? This affects the parameter name choice.